requirements.in

```diff
@@ -24,6 +23,7 @@ rollbar==0.16.2
 Sphinx==4.3.1
 sphinx-rtd-theme==1.0.0
 sphinx-sitemap==2.2.0
+time-machine
```

Comment:
I don't understand why you pin stuff in requirements.in, I recommend not doing so, apart from large rangse like `django>=3.2,<4`, and leaving the pins for requirements.txt only.

Reply:
Thanks for challenging my assumption here. I had to think through why I like to pin my stuff in `requirements.in`. I think it comes down to control.

There are times when I'm not ready to upgrade to a certain package, so I'll pin in the `.in` file. Since that's a possibility and I have slightly obsessive tendencies, I pin them all for consistency.

Dependabot is pip-tools aware so it knows how to update both my `requirements.in` and `requirements.txt` files. I think that means that the end result is roughly the same. There's probably a bit more line noise in my dependabot PR diffs.

Reply:
Fair enough. I didn't know dependabot did that.